### PR TITLE
Add ManualDocument model and upload script

### PR DESCRIPTION
**Pull request by jodizzle**
Resolves https://github.com/opensyllabus/osp-pipeline/issues/147.

- Implements a MiscellaneousDocument model based on the approach where the filenames of the documents are md5sums.  The model and corresponding `load_*` job is a simpler version of the BenningtonDocument model; no entry in `sources` is required because the model is just two entries, `data_md5` and `data`, and correspondingly, `MiscellaneousDocument.from_source` is very straightforward.
- Adds a script for extracting files from a zipfile and saving each in either an S3 bucket or a local directory.  Files are saved with their md5sums as the filename, and stored flatly (no subdirectories).

I've already used the script to upload the two current batches of miscellaneous documents to the bucket "osp-miscellaneous-docs".

The tests for this are pretty limited at the moment.  I've updated the `merge_documents.py` tests accordingly, but with the current approach, testing `load_miscellaneous_documents.py` is blocked in the same way that testing the other `load_*` jobs is blocked.  I could build around it (and I have kind of done some of that work already in the classifier module tests), but it might be more expedient to just try running the job on a cluster and seeing if it works.

Question: Should the model be named MiscDocuments or something instead of MiscellaneousDocuments?  I wanted to be clear and "proper", but I realize that 'miscellaneous' is kind of hard to spell...

### CODE REVIEW

**Comment by jodizzle** [comment]
@wearpants I changed the names to be based on 'manual' instead of 'miscellaneous', as discussed.  I don't really like that name, but I couldn't think of another name that felt both descriptive and convenient to use (assorted docs?  random docs?  md5 docs?).

**Comment by jodizzle** [comment]
Also, I can spin up a cluster and try running the job before merging if that's appropriate.

File: bin/upload_manual_docs.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+from osp_pipeline.corpus.utils import get_md5
+
+
+@click.command()
+@click.argument("zip_path", type=click.Path(exists=True))
+@click.argument("output_dir", type=click.Path())
+def main(zip_path, output_dir):
+    """Extract files from a ZIP archive and save them md5sums as filenames.
+
+    \b
```
Comment (wearpants) [inline bin/upload_manual_docs.py]: what is this \b for?

Comment (jodizzle) [reply bin/upload_manual_docs.py]: If you try to do `./upload_manual_docs.py --help` without the `\b`, the extra indentation and newlines in the docstring (underneath ZIP_PATH and OUTPUT_DIR) don't print out correctly; they just appear as extra spaces.

**Comment by jodizzle** [comment]
I ran and the job and verified it against extractions in a local directory.  After one small fix everything seems to be working.

@wearpants One thing I thought of is the `rows_per_output_partition` class variable.  I just copy-pasted the one from BenningtonDocument, but should it be any different?

**Comment by wearpants** [comment]
How many manual docs are we talking about?

**Comment by jodizzle** [comment]
At the moment there are 17379.

**Comment by wearpants** [comment]
Seems fine

**Comment by jodizzle** [comment]
Alright, gonna go ahead and merge.

**Pull request merged**